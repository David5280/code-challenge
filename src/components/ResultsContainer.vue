<template>
  <div class='results-container'>
    <div class='result-container'>
      <p>Tip:</p>
      <p class='tip-display' v-if='results.tipAmount'>
        ${{ results.tipAmount | numeral('0,0.00') }}
      </p>
    </div>
    <div class='result-container'>
      <p>Total:</p>
      <p class='bill-display' v-if='results.finalBillAmount'>
        ${{ results.finalBillAmount | numeral('0,0.00')  }}
      </p>
    </div>
    <div 
      class='result-container per-person-display' 
      v-if='results.pricePerPerson'
    >
      <p>Per Person:</p>
      <p>${{ results.pricePerPerson | numeral('0,0.00')  }}</p>
    </div>
  </div>
</template>

<script>
export default {
  name: 'ResultsContainer',
  props: {
    results: Object
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
  .results-container {
    align-items: flex-start;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
  }
  .result-container {
    display: flex;
    height: 30px;
    justify-content: space-between;
    width: 100%;
  }
</style>
