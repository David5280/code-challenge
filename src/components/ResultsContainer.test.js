import { mount } from '@vue/test-utils';
import ResultsContainer from './ResultsContainer';

describe('ResultsContainer', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(ResultsContainer, {
      propsData: {
        results: {
          tipAmount: '.20',
          finalBillAmount: '1000'
        }
      }
    });
    expect(wrapper.isVueInstance()).toBeTruthy()
  });
});