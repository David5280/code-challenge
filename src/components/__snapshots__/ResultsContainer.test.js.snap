// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsContainer should match the snapshot 1`] = `
<div
  class="results-container"
>
  <div
    class="result-container"
  >
    <p>
      Tip:
    </p>
     
    <p
      class="tip-display"
    >
      
      $200
    
    </p>
  </div>
   
  <div
    class="result-container"
  >
    <p>
      Total:
    </p>
     
    <p
      class="bill-display"
    >
      
      $1000
    
    </p>
  </div>
   
  <!---->
</div>
`;
